### PR TITLE
Fix fallbacks to OpenGL

### PR DESCRIPTION
**Pull request by Summersay415**
Take in account definition of GLES3_ENABLED
Fix fallback logic on Android

### CODE REVIEW

File: platform/android/java/lib/src/org/godotengine/godot/Godot.kt [hunk 1]
```diff
@@ -477,10 +477,15 @@ class Godot(private val context: Context) {
 			// ...add to FrameLayout
 			containerLayout?.addView(editText)
 			renderView = if (usesVulkan()) {
-				if (!meetsVulkanRequirements(activity.packageManager)) {
+				if (meetsVulkanRequirements(activity.packageManager)) {
+					GodotVulkanRenderView(host, this, godotInputHandler)
+				} else if (canFallbackToOpenGL()) {
+					// Fallback to openGl
```
Comment (AThousandShips) [inline platform/android/java/lib/src/org/godotengine/godot/Godot.kt:483]:
```suggestion
					// Fallback to OpenGl.
```

**Review by m4gr3d — APPROVED** [review]
The Android section looks good!

File: platform/android/display_server_android.cpp [hunk 2]
```diff
@@ -624,19 +618,25 @@ DisplayServerAndroid::DisplayServerAndroid(const String &p_rendering_driver, Dis
...
 				OS::get_singleton()->set_current_rendering_method("gl_compatibility");
 				OS::get_singleton()->set_current_rendering_driver_name(rendering_driver);
 			} else {
+#endif
 				ERR_PRINT(vformat("Failed to initialize %s context", rendering_driver));
 				r_error = ERR_UNAVAILABLE;
 				return;
+#if defined(GLES3_ENABLED)
 			}
+#endif
```
Comment (AThousandShips) [inline platform/android/display_server_android.cpp:635]:
```suggestion
 			} else
#endif
 			{
 				ERR_PRINT(vformat("Failed to initialize %s context", rendering_driver));
 				r_error = ERR_UNAVAILABLE;
 				return;
 			}
```
Cleaner

File: platform/linuxbsd/wayland/display_server_wayland.cpp [hunk 3]
```diff
@@ -1356,13 +1356,15 @@ DisplayServerWayland::DisplayServerWayland(const String &p_rendering_driver, Win
...
 			rendering_context = nullptr;
+#if defined(GLES3_ENABLED)
 			bool fallback_to_opengl3 = GLOBAL_GET("rendering/rendering_device/fallback_to_opengl3");
 			if (fallback_to_opengl3 && rendering_driver != "opengl3") {
 				WARN_PRINT("Your video card drivers seem not to support the required Vulkan version, switching to OpenGL 3.");
 				rendering_driver = "opengl3";
 				OS::get_singleton()->set_current_rendering_method("gl_compatibility");
 				OS::get_singleton()->set_current_rendering_driver_name(rendering_driver);
 			} else {
+#endif // GLES3_ENABLED
```
Comment (AThousandShips) [inline platform/linuxbsd/wayland/display_server_wayland.cpp:1367]:
```suggestion
 			} else
#endif // GLES3_ENABLED
 			{
```

File: platform/linuxbsd/wayland/display_server_wayland.cpp [hunk 4]
```diff
@@ -1377,7 +1379,9 @@ DisplayServerWayland::DisplayServerWayland(const String &p_rendering_driver, Win
 				}
 
 				ERR_FAIL_MSG(vformat("Could not initialize %s", rendering_driver));
+#if defined(GLES3_ENABLED)
 			}
+#endif // GLES3_ENABLED
```
Comment (AThousandShips) [inline platform/linuxbsd/wayland/display_server_wayland.cpp:1384]:
```suggestion
 			}
```

File: platform/linuxbsd/x11/display_server_x11.cpp [hunk 5]
```diff
@@ -6168,13 +6168,15 @@ DisplayServerX11::DisplayServerX11(const String &p_rendering_driver, WindowMode
...
 			rendering_context = nullptr;
+#if defined(GLES3_ENABLED)
 			bool fallback_to_opengl3 = GLOBAL_GET("rendering/rendering_device/fallback_to_opengl3");
 			if (fallback_to_opengl3 && rendering_driver != "opengl3") {
 				WARN_PRINT("Your video card drivers seem not to support the required Vulkan version, switching to OpenGL 3.");
 				rendering_driver = "opengl3";
 				OS::get_singleton()->set_current_rendering_method("gl_compatibility");
 				OS::get_singleton()->set_current_rendering_driver_name(rendering_driver);
 			} else {
+#endif // GLES3_ENABLED
```
Comment (AThousandShips) [inline platform/linuxbsd/x11/display_server_x11.cpp:6179]:
```suggestion
 			} else
#endif // GLES3_ENABLED
 			{
```

File: platform/linuxbsd/x11/display_server_x11.cpp [hunk 6]
```diff
@@ -6189,7 +6191,9 @@ DisplayServerX11::DisplayServerX11(const String &p_rendering_driver, WindowMode
 				}
 
 				ERR_FAIL_MSG(vformat("Could not initialize %s", rendering_driver));
+#if defined(GLES3_ENABLED)
 			}
+#endif // GLES3_ENABLED
```
Comment (AThousandShips) [inline platform/linuxbsd/x11/display_server_x11.cpp:6196]:
```suggestion
 			}
```

File: platform/macos/display_server_macos.mm [hunk 7]
```diff
@@ -3623,6 +3623,40 @@
...
+				rendering_driver = "opengl3";
+				OS::get_singleton()->set_current_rendering_method("gl_compatibility");
+				OS::get_singleton()->set_current_rendering_driver_name(rendering_driver);
+			} else {
+#endif
+				r_error = ERR_CANT_CREATE;
+				ERR_FAIL_MSG("Could not initialize " + rendering_driver);
+#if defined(GLES3_ENABLED)
+			}
+#endif
```
Comment (AThousandShips) [inline platform/macos/display_server_macos.mm:3655]:
```suggestion
			} else
#endif
			{
				r_error = ERR_CANT_CREATE;
				ERR_FAIL_MSG("Could not initialize " + rendering_driver);
			}
```

File: platform/ios/display_server_ios.mm [hunk 8]
```diff
@@ -111,19 +111,25 @@
...
 				OS::get_singleton()->set_current_rendering_method("gl_compatibility");
 				OS::get_singleton()->set_current_rendering_driver_name(rendering_driver);
 			} else {
+#endif
 				ERR_PRINT(vformat("Failed to initialize %s context", rendering_driver));
 				r_error = ERR_UNAVAILABLE;
 				return;
+#if defined(GLES3_ENABLED)
 			}
+#endif
```
Comment (AThousandShips) [inline platform/ios/display_server_ios.mm:128]:
```suggestion
 			} else
#endif
 			{
 				ERR_PRINT(vformat("Failed to initialize %s context", rendering_driver));
 				r_error = ERR_UNAVAILABLE;
 				return;
 			}
```

**Review by bruvzg — APPROVED** [review]

**Comment by Repiteo** [comment]
Thanks!

**Pull request merged**